### PR TITLE
Lua cvar library

### PR DESCRIPTION
**Pull request by viviicat**

### CODE REVIEW

File: src/scriptsys/scr_cvar.c [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+#include "lua/lauxlib.h"
+#include "math/types.h"
+#include "common/console.h"
+#include "common/cvar.h"
+#include "scr_cmd.h"
+#include "script.h"
+
+static void scr_push_vec(lua_State* L, float4 vec)
+{
+	lua_createtable(L, 4, 0);
```
Comment (gheshu) [inline src/scriptsys/scr_cvar.c:12]: table == {} or an object in lua, right?
A TODO for later: we should figure out a faster way to pass structs to lua, maybe via FFI, or maybe just push 4 floats?
You may want to use VEC_CALL on anything taking or returning a math type like float4 (unless its passed as a function pointer to lua).

File: src/scriptsys/scr_cvar.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include "script.h"
+#include "common/macro.h"
+
+PIM_C_BEGIN
+
+void scr_cvar_init(lua_State* L);
```
Comment (gheshu) [inline src/scriptsys/scr_cvar.h:6]: Would be good to have a matching shutdown for any init function, so that we can hot reload sections of the engine.

**Review by Vethanis — APPROVED** [review]
Cool!

File: src/scriptsys/scr_cvar.c (hunk 1 above)
Comment (viviicat) [reply src/scriptsys/scr_cvar.c:12]: I think we could use `UserData` for this, and then have the X, Y, Z, W be fetched using metatables. Would have to profile performance!

Comment (viviicat) [reply src/scriptsys/scr_cvar.c:12]: #23

**Pull request merged**